### PR TITLE
Improve json schema build

### PR DESCRIPTION
**Pull request by przepompownia**
- introduce enums
- change schema URL to working with vscode json ls
- use types
- generate default values
- add example usage

### CODE REVIEW

**Comment by przepompownia** [comment]
Why not to generate defaults?

**Comment by przepompownia** [comment]
When I opened autogenerated JSON schema to manual edition, completion did not work with the original URL so I changed it to some working.

File: composer.json [hunk 1]
```diff
@@ -78,6 +78,18 @@
         "phpactor/worse-reference-finder": "0.2.6",
         "phpactor/worse-reflection-extension": "0.2.5"
     },
+    "repositories": [
+        {
+            "type": "vcs",
+            "no-api": true,
+            "url": "https://github.com/przepompownia/container"
```
Comment (dantleech) [inline composer.json:85]: is there a PR for container also?

File: lib/Extension/Core/Model/JsonSchemaBuilder.php [hunk 2]
```diff
@@ -28,7 +28,7 @@ public function __construct(string $title, array $extensions)
     public function dump(): string
     {
         $schema = [
-          '$schema' => 'https =>//json-schema.org/draft/2020-12/schema',
+          '$schema' => 'https://json-schema.org/draft-07/schema',
```
Comment (dantleech) [inline lib/Extension/Core/Model/JsonSchemaBuilder.php:31]: i think the original is valid without the `=>` typo: https://json-schema.org/learn/getting-started-step-by-step

**Review by dantleech — APPROVED** [review]
nice. the enum is definitely something I wanted to add, i use a similar approach in PHPBench using the options Symfony resolver:

```
                    $values = $inspector->getAllowedValues($option);
                    $meta['enum'] = $values;
```

**Comment by dantleech** [comment]
the map resolver PR is failing

File: composer.json (hunk 1 above)
Comment (przepompownia) [reply composer.json:85]: https://github.com/phpactor/container/pull/12

Comment (dantleech) [reply composer.json:85]: cool closed that, we can now just require `map-resolver: ^1.5.0`

File: lib/Extension/Core/Model/JsonSchemaBuilder.php (hunk 2 above)
Comment (przepompownia) [reply lib/Extension/Core/Model/JsonSchemaBuilder.php:31]: Probably it's valid but it does not work for me for some reason, in opposite to `draft-07`. Maybe it's a problem of vscode-json-language-server. It reports that this draft it not fully supported yet. 

It's not important as we do not edit the schema file manually in normal case. If you prefer `2020-12` then no problem for me to change it back.

Comment (dantleech) [reply lib/Extension/Core/Model/JsonSchemaBuilder.php:31]: no it's fine i guess

**Comment by przepompownia** [comment]
https://github.com/phpactor/phpactor/runs/6173691607?check_suite_focus=true Could you look over there? I have no permission to run the build again.

**Comment by dantleech** [comment]
i changed the settings for phpactor so that the build runs for non-new GH users

**Comment by przepompownia** [comment]
Thanks!

**Pull request merged**